File: MongoBooks2/JavaScript/Node/AngularCLI/my-project/src/app/Models/Book.ts
import IBook = books.IBook;

export class Book implements IBook {
  public _id: string;
  public dateString: string;
  public date: Date;
  public author: string;
  public title: string;
  public pages: number;
  public note: string;
  public nationality: string;
  public originalLanguage: string;
  public image_url: string;
  public tags: Array<string>;
  public format: number;
  public prettyDate: string;
  public prettyFormat: string;
  
  constructor(
    _id: string,
    dateString: string,
    date: Date,
    author: string,
    title: string,
    pages: number,
    note: string,
    nationality: string,
    originalLanguage: string,
    image_url: string,
    tags: Array<string>,
    format: number) {
    this._id = _id;
    this.dateString = dateString;
    this.date = date;
    this.author = author;
    this.title = title;
    this.pages = pages;
    this.note = note;
    this.nationality = nationality;
    this.originalLanguage = originalLanguage;
    this.image_url = image_url;
    this.tags = tags;
    this.format = format;

    this.prettyDate = this.displayDate();
    this.prettyFormat = this.displayFormat();
  }

  private displayFormat(): string {
    const formats = ["undefined", "Book", "Comic", "Audio"];
    return formats[this.format % formats.length];
  }

  private displayDate(): string {
    const months = ["January", "February", "March", "April", "May", "June",
      "July", "August", "September", "October", "November", "December"];
    const days = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"];

    const weekday = days[this.date.getDay()];
    const displayDay = this.nthDay(this.date.getDate());
    const month = months[this.date.getMonth()];
    return weekday + " " + displayDay + " " + month + " " + this.date.getFullYear();
  }

  private nthDay(day: number): string {
    switch (day % 10) {
    case 1:
      return day + "st";
    case 2:
      return day + "nd";
    case 3:
      return day + "rd";
    default:
      return day + "th";
    }
  }
}

File: MongoBooks2/AngularMongoBooks3/Controllers/DataClasses/Book.cs
namespace AngularMongoBooks3.Controllers.DataClasses
{
    using System;


    using BooksCore.Books;

    public class Book
    {
        /// <summary>
        /// Gets or sets the date string.
        /// </summary>
        public string DateString { get; set; }

        /// <summary>
        /// Gets or sets the date.
        /// </summary>
        public DateTime Date { get; set; }

        /// <summary>
        /// Gets or sets the author.
        /// </summary>
        public string Author { get; set; }

        /// <summary>
        /// Gets or sets the title.
        /// </summary>
        public string Title { get; set; }

        /// <summary>
        /// Gets or sets the pages.
        /// </summary>
        public int Pages { get; set; }

        /// <summary>
        /// Gets or sets the note.
        /// </summary>
        public string Note { get; set; }

        /// <summary>
        /// Gets or sets the nationality.
        /// </summary>
        public string Nationality { get; set; }

        /// <summary>
        /// Gets or sets the original language.
        /// </summary>
        public string OriginalLanguage { get; set; }

        /// <summary>
        /// Gets or sets the image URL.
        /// </summary>
        public string ImageUrl { get; set; }

        /// <summary>
        /// Gets or sets the tags.
        /// </summary>
        public string[] Tags { get; set; }

        /// <summary>
        /// Gets or sets the user name.
        /// </summary>
        public string User { get; set; }

        /// <summary>
        /// Gets or sets the book format.
        /// </summary>
        public string Format { get; set; }

        public Book()
        {

        }

        public Book(BookRead book)
        {
            DateString = book.DateString;
            Date = book.Date;
            Author = book.Author;
            Title = book.Title;
            Pages = book.Pages;
            Note = book.Note;
            Nationality = book.Nationality;
            OriginalLanguage = book.OriginalLanguage;
            ImageUrl = book.ImageUrl;
            User = book.User;
            Format = book.Format.ToString();
            Tags = new string[book.Tags.Count];
            for (int i = 0; i < book.Tags.Count; i++)
            {
                Tags[i] = book.Tags[i];
            }
        }

        //public GetBookRead()

    }
}
